ironfish-zkp/src/circuits/output.rs

```diff
@@ -0,0 +1,265 @@
+use ff::PrimeField;
+use group::Curve;
+
+use bellman::{Circuit, ConstraintSystem, SynthesisError};
+
+use jubjub::{ExtendedPoint, SubgroupPoint};
+use zcash_primitives::sapling::ValueCommitment;
+
+use zcash_proofs::{
+    circuit::{
+        ecc::{self},
+        pedersen_hash,
+    },
+    constants::NOTE_COMMITMENT_RANDOMNESS_GENERATOR,
+};
+
+use super::util::expose_value_commitment;
+use bellman::gadgets::boolean;
+
+pub const PUBLIC_KEY_GENERATOR: SubgroupPoint = SubgroupPoint::from_raw_unchecked(
```

Comment:
Can we move this to [`constants.rs`](https://github.com/iron-fish/ironfish/blob/network-reset/ironfish-zkp/src/constants.rs)?